pkg/controller/handler.go

```diff
@@ -12,41 +12,85 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 )
 
+const SAS_PREFIX = "k8scc01covidacr.azurecr.io/sas:"
+const FEATURES_LABEL = "state.aaw.statcan.gc.ca/employee-only-features"
+const RB_LABEL = "state.aaw.statcan.gc.ca/non-employee-user"
+
 func sasImage(pod *corev1.Pod) bool {
-	image := pod.Spec.Containers[0].Image
-	sasImage := strings.HasPrefix(image, "k8scc01covidacr.azurecr.io/sas:")
-	return sasImage
+
+	for _, container := range pod.Spec.Containers {
+		sasImage := strings.HasPrefix(container.Image, SAS_PREFIX)
+		if sasImage {
+			return true
+		}
+	}
+
+	return false
 }
 
-func (c *Controller) handleProfile(profile *v1.Profile) error {
-	ctx := context.Background()
+func isEmployee(email string) bool {
+
+	employeeDomains := [2]string{"cloud.statcan.ca", "statcan.gc.ca"} // Intialized with values
+	for _, domain := range employeeDomains {
+		if strings.HasSuffix(email, domain) {
+			return true
+		}
+	}
+	return false
+}
 
+func (c *Controller) handleProfile(profile *v1.Profile) error {
 	namespace := profile.Name
 	hasEmpOnlyFeatures := false
 
+	// label set from rolebindings, if there are no rolebindings, then the default is that they are external
+	nonEmployeeUser := true
+
 	pods, err := c.podLister.Pods(namespace).List(labels.Everything())
+
+	if err != nil {
+		return err
+	}
+
+	roleBindings, err := c.roleBindingLister.RoleBindings(namespace).List(labels.Everything())
+
 	if err != nil {
 		return err
 	}
+
 	for _, pod := range pods {
 		if sasImage(pod) {
 			hasEmpOnlyFeatures = true
 			break
 		}
 	}
 
+	for _, roleBindings := range roleBindings {
```

Comment:
The logic here could permit a non-employee to gain access to a SAS image. For example, suppose a non employee and an employee are in the same namespace. There would exist rolebindings associated with both the employee and non-employee, and the code below would set `nonEmployeeUser = false` if it detects the employee's rolebinding.

Instead this check should go in the opposite direction.

Assume `nonEmployeeUser := false` and if there exists a single counterexample, set `nonEmployeeUser := true` .


Example test case:

```yaml
kind: RoleBinding
# ...
subjects:
- name: alice@statcan.gc.ca
----
kind: RoleBinding
# ...
subjects:
- name: hacker@malicious.com
```